core/io/stream_peer_tcp.cpp

```diff
@@ -80,6 +82,12 @@ Error StreamPeerTCP::connect_to_host(const IP_Address &p_host, uint16_t p_port)
 
 	_sock->set_blocking_enabled(false);
 
+	if (_sock->bind(IP_Address("*"), p_local_port) != OK) {
```

Comment:
I can foresee this giving us troubles in some dual stack networks.
If we want to be able to bind before connect in TCP (allowing for those rare cases where TCP NAT-traversal works), I think it's safer to expose it as a separate `bind_socket` method that returns an `Error`.